presto-orc/src/main/java/com/facebook/presto/orc/cache/StorageOrcFileTailSource.java

```diff
@@ -92,9 +108,18 @@ public OrcFileTail getOrcFileTail(OrcDataSource orcDataSource, MetadataReader me
         int footerSize = toIntExact(postScript.getFooterLength());
         int metadataSize = toIntExact(postScript.getMetadataLength());
 
+        boolean readDwrfStripeCache = dwrfStripeCacheEnabled
+                && postScript.getDwrfStripeCacheLength().isPresent()
+                && postScript.getDwrfStripeCacheMode().isPresent();
```

Comment:
What happens if stripeCacheLength is set to negative or cache mode is set to None.

Reply:
Q1: What happens if stripeCacheLength is set to negative?
A1: We would fail down the line, same as if footerSize or metadataSize were set to negative. I can add a bunch of `>0` checks for these 3 fields.

Q2. What happens if cache mode is set to None.
A2: It depends. `None` means that either the cache was not used or that there is a new unsupported cache mode. If the cache was not used and the rest of fields are present we would have an empty cache. If an unsupported mode was used we would and the rest of the cache fields have valid values then we would have an unusable cache.
Do you suggest to proceed only if the cache mode is != NONE?

Reply:
I did not want to change the code. I like the code as it is.  But want to get your opinions  on it.

Let us made all the new checks only when stripeCacheMode is enabled. If Stripe Cache is corrupted, previously Presto ignored and did not fail. We want it to fail in the new path, but when disabled it should still pass.

Reply:
Updated the PR:
- added checks for sizes to not be negative
- avoid loading the stripe cache if mode is NONE

Reply:
The stripe cache size check will be triggered only if this feature is enabled, so we should be good here.

Adding checks for the metadataSize and footerSize just makes these checks more verbose. If I understand it correctly right now these checks are implicitly executed when we attempt to create a slice.